apps/ewallet/lib/ewallet/validators/transaction_consumption_validator.ex

```diff
@@ -35,14 +35,12 @@ defmodule EWallet.TransactionConsumptionValidator do
           | {:error, Atom.t()}
   def validate_before_confirmation(consumption, owner) do
     with {request, balance} <- {consumption.transaction_request, consumption.balance},
-         true <-
-           TransactionRequest.is_owned_by?(request, owner) || :not_transaction_request_owner,
          true <-
            TransactionRequest.is_owned_by?(request, owner) || :not_transaction_request_owner,
          {:ok, request} <- TransactionRequest.expire_if_past_expiration_date(request),
+         {:ok, _balance} <- validate_max_consumptions_per_user(request, balance),
          true <- TransactionRequest.valid?(request) || request.expiration_reason,
-         {:ok, consumption} = TransactionConsumption.expire_if_past_expiration_date(consumption),
-         {:ok, _balance} <- validate_max_consumptions_per_user(request, balance) do
+         {:ok, consumption} = TransactionConsumption.expire_if_past_expiration_date(consumption) do
```

Comment:
Curious why the shuffle?

Reply:
Hmmm, because the `max_consumptions_per_user` is a transaction request check, it made more sense to have it before reaching the validation for the consumption (especially since the following line checks if the consumption got expired or not). The `TransactionConsumption.expire_if_past_expiration_date` will actually never return an error so it should be the last line of the `with` :)